lib/optimize/RemoveParentModulesPlugin.js

```diff
@@ -58,12 +61,17 @@ class RemoveParentModulesPlugin {
 	apply(compiler) {
 		compiler.plugin("compilation", (compilation) => {
 			compilation.plugin(["optimize-chunks-basic", "optimize-extracted-chunks-basic"], (chunks) => {
-				chunks.forEach((chunk) => {
-					const cache = {};
-					chunk.modules.slice().forEach((module) => {
-						if(chunk.parents.length === 0) return;
-						const dId = "$" + debugIds(module.chunks);
-						let parentChunksWithModule;
+				for(var index = 0, lenChunks = chunks.length; index < lenChunks; index++) {
+					var chunk = chunks[index];
+
+					var cache = {};
```

Comment:
I checked the speed and I got bad results for Map.

> Map x 417 ops/sec ±1.76% (82 runs sampled)
> Native x 8,021 ops/sec ±1.69% (84 runs sampled)

https://gist.github.com/Tom910/8642ee4d9a4913cd40d1581ab3d95738

For this reason, it is better not to rewrite Map.